docs/DeveloperGuide.md

```diff
@@ -1,15 +1,596 @@
+
+After `run()` is called by `main()` in Duke, the `CommandParser` is first called to parse the command. The initial
+ input is split into the commandWord and parameters using `splitInput()`. Next, the command word is parsed using 
+ `getCommand()`. With the new command created, the parameters are then parsed by calling `ParameterParser.parse()`
+ 
+In `ParameterParser`, multiple checks are done to ensure that the userInput is accurate. For example, checks are done 
+to confirm that all mandatory tags are filled, that no unsupported tags are used, no duplicate tags, no tags are
+without parameters and finally that the formats of parameters are accurate. After all the checks are done`setCommand` 
+is called to customize the command accordingly.
+
+With the checked and accurate commands and parameters, the command is then executed by `Duke`
+
+_Written by: Paul Low_
+
+### UI Component
+
+The UI component consists of a `Ui` class that displays information and error messages based on the user 
+input and the behavior of the application. Static messages are pre-defined in the `ErrorMessages` and `InfoMessages` 
+classes from the Common component, while  dynamic messages such as a transactions list may be generated during 
+execution of the application.
+
+<p align="center">
+    <img src="images/UiComponentClassDiagram.png">
+    <br />
+    <i>Figure 2.6: Class Diagram for UI Component</i>
+</p>
+
+As seen from the class diagram, every command that requires the ability to print to the system output will have
+to call the functions from the `Ui` class. To add on, the `Duke` class will also use the `Ui` class to read user 
+input.
+
+_Written by: Chua Han Yong Darren_
+
+### Common Component
+
+_Written by: Author name_
+
+## Implementation
+
+### Overview for Transaction
+
+The following commands are accepted in Moolah Manager:
+1) `Add` - Adds an `Income` or `expense` type transaction into the list of transaction. [Add Command](#add-command)
+2) `Budget` - Adds an `Budget` into Moolah Manager, which sets the basis for financial tracking. [Budget Command](#budget-command)
+3) `Bye` - Exits Moolah Manager. [Bye Command](#bye-command)
+4) `Delete` - Deletes an `Income` or `expense` type transaction from the list of transaction. [Delete Command](#delete-command)
+5) `Edit` - Edits an `Income` or `expense` type transaction from the list of transaction. [Edit Command](#edit-command)
+6) `Find` - Searches for an `Income` or `expense` type transaction from the list of transaction given keywords. [Find Command](#find-command)
+7) `Help` - Outputs the usage of commands of Moolah Manager. [Help Command](#help-command)
+8) `List` - List all transactions. [List Command](#list-command)
+9) `Purge` - Deletes all transactions from the list of transaction. [Purge Command](#purge-command)
+10) `Stats` - View statistics on transactions based on the list of transactions in Moolah Manager. [Stats Command](#stats-command)
```

Comment:
I believe the numbered list may not be rendered correctly on the site.

Reply:
Indeed it is not rendered correctly, we will have to amend this. @brian-vb do note to update the numbering in markdown format (e.g. `1.`, `2.`, ...)